lib/shopify_api/bulk/query.ex

```diff
@@ -169,4 +182,63 @@ defmodule ShopifyAPI.Bulk.Query do
       _ -> poll(token, bulk_query_id, polling_rate, max_poll, depth + 1)
     end
   end
+
+  defp httpoison_streamed_get!(url) do
+    Stream.resource(
+      fn ->
+        HTTPoison.get!(url, %{}, stream_to: self(), async: :once)
+      end,
+      fn %HTTPoison.AsyncResponse{id: id} = resp ->
+        receive do
+          %HTTPoison.AsyncStatus{id: ^id, code: 200} ->
+            HTTPoison.stream_next(resp)
+            {[], resp}
+
+          %HTTPoison.AsyncStatus{id: ^id, code: code} ->
+            raise("ShopifyAPI.Bulk stream fetch got non 200 code of: #{code}")
```

Comment:
Might be a good idea to build out an exception module for this corner of the API at some point, since it has slightly different potential failure cases.